### PR TITLE
NMS-13715: Fixed the DB installer for Azure PostgreSQL database

### PR DESCRIPTION
**Pull request by yangli2014**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13715

### CODE REVIEW

**Comment by yangli2014** [comment]
The data source config file should look like the following

`<jdbc-data-source name="opennms"
                    database-name="opennms"
                    class-name="org.postgresql.Driver"
                    url="jdbc:postgresql://yang-opennms.postgres.database.azure.com:5432/opennms?sslmode=require"
                    user-name="opennms@yang-opennms"
                    password="opennms" />  
<jdbc-data-source name="opennms-admin"
                    database-name="template1"
                    class-name="org.postgresql.Driver"
                    url="jdbc:postgresql://yang-opennms.postgres.database.azure.com:5432/template1?sslmode=require"
                    user-name="postgres@yang-opennms"
                    password="N'5J5*vr" />`

File: core/schema/src/main/java/org/opennms/core/schema/Migrator.java [hunk 1]
```diff
@@ -452,14 +452,20 @@ public void createUser() throws MigrationException {
...
-            st.execute("CREATE USER " + getDatabaseUser() + " WITH PASSWORD '" + getDatabasePassword() + "'");
+            st.execute("CREATE USER " + getUserForDDL() + " WITH PASSWORD '" + getDatabasePassword() + "'");
         } catch (final SQLException e) {
             throw new MigrationException("an error occurred creating the OpenNMS user", e);
         } finally {
             cleanUpDatabase(c, null, st, rs);
         }
     }
 
+    protected String getUserForDDL() {
```
Comment (j-white) [inline core/schema/src/main/java/org/opennms/core/schema/Migrator.java:463]: DDL as in Data Definition Language (DDL)?

File: core/schema/src/test/java/org/opennms/core/schema/MigratorTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Before
+    public void setup() {
+        migrator = new Migrator();
+    }
+
+
+    @Test
+    public void testGetUserForDDLWithHostname() {
+        String userName = "opennms@@test-env-onms";
```
Comment (j-white) [inline core/schema/src/test/java/org/opennms/core/schema/MigratorTest.java:48]: Is the duplicate `@` intentional?

**Review by j-white — COMMENTED** [review]
Looks reasonable to me - will keep the existing behavior if no `@` exists in the username.

File: core/schema/src/main/java/org/opennms/core/schema/Migrator.java (hunk 1 above)
Comment (yangli2014) [reply core/schema/src/main/java/org/opennms/core/schema/Migrator.java:463]: Yes

File: core/schema/src/test/java/org/opennms/core/schema/MigratorTest.java (hunk 2 above)
Comment (yangli2014) [reply core/schema/src/test/java/org/opennms/core/schema/MigratorTest.java:48]: Good catch.

**Comment by RangerRick** [comment]
> The data source config file should look like the following

I recommend you change your azure database password right now.  :)

**Review by RangerRick — DISMISSED** [review]

**Review by j-white — APPROVED** [review]

**Pull request merged**